app/jobs/samples/batch_sample_import_job.rb

```diff
@@ -5,47 +5,65 @@ module Samples
-      ::Samples::BatchFileImportService.new(namespace, current_user, blob_id, params).execute
-
-      # TODO: change all these from metadata to batch sample
-      # if namespace.errors.empty?
-      #   Turbo::StreamsChannel.broadcast_replace_to(
-      #     broadcast_target,
-      #     target: 'import_metadata_dialog_content',
-      #     partial: 'shared/samples/metadata/file_imports/success',
-      #     locals: {
-      #       type: :success,
-      #       message: I18n.t('shared.samples.metadata.file_imports.success.description')
-      #     }
-      #   )
-
-      # elsif namespace.errors.include?(:sample)
-      #   errors = namespace.errors.messages_for(:sample)
-
-      #   Turbo::StreamsChannel.broadcast_replace_to(
-      #     broadcast_target,
-      #     target: 'import_metadata_dialog_content',
-      #     partial: 'shared/samples/metadata/file_imports/errors',
-      #     locals: {
-      #       type: :alert,
-      #       message: I18n.t('shared.samples.metadata.file_imports.errors.description'),
-      #       errors: errors
-      #     }
-      #   )
-      # else
-      #   errors = namespace.errors.full_messages_for(:base)
-
-      #   Turbo::StreamsChannel.broadcast_replace_to(
-      #     broadcast_target,
-      #     target: 'import_metadata_dialog_content',
-      #     partial: 'shared/samples/metadata/file_imports/errors',
-      #     locals: {
-      #       type: :alert,
-      #       errors: errors
-      #     }
-      #   )
-      # end
+    def perform(namespace, current_user, broadcast_target, blob_id, params) # rubocop:disable Metrics/MethodLength
+      response = ::Samples::BatchFileImportService.new(namespace, current_user, blob_id, params).execute(
+        Flipper.enabled?(:progress_bars) ? broadcast_target : nil
+      )
+
+      if namespace.errors.empty?
+        handle_success(broadcast_target, response)
+
+      elsif namespace.errors.include?(:sample)
+        errors = namespace.errors.messages_for(:sample)
+
+        Turbo::StreamsChannel.broadcast_replace_to(
+          broadcast_target,
+          target: 'import_spreadsheet_dialog_content',
+          partial: 'shared/samples/spreadsheet_imports/errors',
+          locals: {
+            type: :alert,
+            message: I18n.t('shared.samples.spreadsheet_imports.errors.description'),
+            errors: errors
+          }
+        )
+      else
+        errors = namespace.errors.full_messages_for(:base)
+
+        Turbo::StreamsChannel.broadcast_replace_to(
+          broadcast_target,
+          target: 'import_spreadsheet_dialog_content',
+          partial: 'shared/samples/spreadsheet_imports/errors',
+          locals: {
+            type: :alert,
+            errors: errors
+          }
+        )
+      end
+    end
+
+    private
+
+    def handle_success(broadcast_target, response) # rubocop:disable Metrics/MethodLength
```

Comment:
I think this method should be renamed to `handle_response` as this doesn't just handle a successful import, but handles any of the issues encountered as well

Reply:
Fixed in https://github.com/phac-nml/irida-next/pull/984/commits/1c30d7b0e5a3d125104197aab4e8b2066f557f74